### DISCUSSION TITLE
<sequenceReadsPaired> input generates a separate invocation for each pair

### DISCUSSION THREAD

**Discussion by innovate-invent** [phac-nml/irida#1344] [Ideas]
## Describe the bug
<sequenceReadsPaired> input generates a separate invocation for each pair
IRIDA always creates a list:pair with a single pair in the list

## Steps to reproduce the problem
What were you doing when you encountered the problem?
1. Add two samples to cart
2. Run RefSeqMasher Pipeline on both samples
3. View analysis list

## Expected behaviour
What did you expect to happen?
A single analysis should be created, passing a list:pair to the workflow.
The list:pair should have two elements, a pair for each sample.

## Additional context
Additional information to help solve the problem.  Test files, screenshots, IRIDA version information can go here.

**Comment by apetkau** [comment]
Thank you for reporting @innovate-invent, I appreciate it.

I'm not quite sure what you mean for this bug though, since this sounds like it's working as expected. The RefSeqMasher pipeline should submit a separate invocation for each pair. This can be controlled with the `<requiresSingleSample>` element in a workflow XML file - https://phac-nml.github.io/irida-documentation/developer/tools/pipelines/workflow-description/#requiressinglesample

For example:

```xml
<!-- Used when you wish to invoke a single Galaxy workflow instance for all samples -->
<requiresSingleSample>false</requiresSingleSample>

<!-- Used when you wish to invoke a separate Galaxy workflow instance for every sample-->
<requiresSingleSample>true</requiresSingleSample>
```

**Comment by innovate-invent** [comment]
The RefSeqMasher workflow expects a `list:pair` implying that all samples should be handed to a single invocation.
https://github.com/phac-nml/irida/blob/181b427ee1f2afb3f5becc81a80c57fb765b9657/src/main/resources/ca/corefacility/bioinformatics/irida/model/workflow/analysis/type/workflows/RefSeqMasherOnPairedReads/0.1/irida_workflow_structure.ga#L27

The workflow should expect a `pair` if the intent is a single invocation per sample. `<requiresSingleSample>true</requiresSingleSample>` should cause a `pair` to be created instead of `list:pair`.

https://github.com/phac-nml/irida/blob/181b427ee1f2afb3f5becc81a80c57fb765b9657/src/main/resources/ca/corefacility/bioinformatics/irida/model/workflow/analysis/type/workflows/RefSeqMasherOnPairedReads/0.1/irida_workflow.xml#L8-L9

**Comment by apetkau** [comment]
Ah, okay. I understand. Yes, I can see how that can be confusing since `list:pair` implies an arbitrarily sized list.

I will relabel this as a `request` though since this was how it was designed from the beginning. The idea being that you use a list of paired-end files with a single element for pipelines that create a single invocation per sample.
